signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/view/SignInWebViewDialogFragment.kt

```diff
@@ -0,0 +1,124 @@
+package com.willowtreeapps.signinwithapplebutton.view
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.webkit.WebView
+import androidx.fragment.app.DialogFragment
+import com.willowtreeapps.signinwithapplebutton.SignInWithAppleCallback
+import com.willowtreeapps.signinwithapplebutton.SignInWithAppleService
+
+@SuppressLint("SetJavaScriptEnabled")
+internal class SignInWebViewDialogFragment : DialogFragment, SignInWithAppleCallback {
+
+    private var authenticationAttempt: SignInWithAppleService.AuthenticationAttempt? = null
+    private var callback: SignInWithAppleCallback? = null
+
+    private val webViewIfCreated: WebView?
+        get() = view as? WebView
+
+    constructor() : super() {
+        authenticationAttempt = null
+    }
+
+    constructor(authenticationAttempt: SignInWithAppleService.AuthenticationAttempt) {
+        this.authenticationAttempt = authenticationAttempt
+    }
+
+    fun configure(
+        callback: SignInWithAppleCallback
+    ) {
+        this.callback = callback
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        if (authenticationAttempt == null) {
+            authenticationAttempt = savedInstanceState?.getParcelable("authenticationAttempt")
```

Comment:
Might want to make `authenticationAttempt` a variable in a companion object since it's being referenced multiple times

Reply:
I thought companion objects were for static members, but that's a shallow understanding. How would a companion object solve this?